### PR TITLE
Support proxy settings for Aether Module resolver

### LINKED ISSUE #272: AetherModuleResolver may not be picking up proxy settings
As a user, I'm deploying a `stream` behind the firewall with an artifact repository that acts as a proxy; however, I'm unable to successfully deploy the stream, as the `AetherModuleResolver` resolver may not be picking up the proxy settings to resolve and download the module jar's required by the stream.

**Acceptance:**
* Module jar's are successfully resolved by `AetherModuleResolver`
* Streams successfully deploy behind the firewall with an artifact repository that acts as a proxy

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add `AetherProxyProperties` as a ConfigurationProperties for the proxy settings needed for the ProxySelector in Aether system
 - Update proxy settings in two places:
   1) Where the remote repositories are setup
   2) WHere the repository system session is created to resolve the artifacts

This resolves #272

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
I couldn't get to test this change behind proxy/firewall settings yet. I tried to setup local proxy but not progressing much. 
I would like to submit this PR so that someone could test this out against proxy/firewall settings.

The proxy properties introduced in this PR have `maven.proxy` as the prefix.

File: spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java [hunk 1]
```diff
@@ -133,6 +182,13 @@ private DefaultRepositorySystemSession newRepositorySystemSession(RepositorySyst
 		LocalRepository localRepo = new LocalRepository(localRepoPath);
 		session.setLocalRepositoryManager(system.newLocalRepositoryManager(session, localRepo));
 		session.setOffline(this.offline);
+		if (isProxyEnabled()) {
+			DefaultProxySelector proxySelector = new DefaultProxySelector();
+			Proxy proxy = new Proxy(proxyProperties.getProtocol(), proxyProperties.getHost(),
+					Integer.parseInt(proxyProperties.getPort()), authentication);
```
Comment (ericbottard) [inline spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/resolver/AetherModuleResolver.java]: Why not make the port property an int in the first place?

**Comment by ericbottard** [comment]
Haven't tested either, but I have one question: why is authentication tied to proxy? Seems like we could benefit from authentication on the remote repo directly, no?

**Comment by ilayaperumalg** [comment]
Good point. We can have authentication for remoteRepositories as well as the proxy (that also includes authentication). Only caveat with the remoteRepositories though is that, we need to have credentials mapped for each remote repository values provided. may be we can address that in a separate PR. I will add proxy configuration for the remoteRepositories and address the direct authentication for remote repo in a separate PR.

**Comment by ilayaperumalg** [comment]
Pushed the changes that sets the authentication **only** for proxy settings.

**Comment by mbogoevici** [comment]
Rebased, squashed, merged w/https://github.com/spring-cloud/spring-cloud-stream/commit/b98d91da394d4d98dbf08e3cb1bd30455825c76d

**Pull request closed without merging**